In [1]:
import pandas as pd

In [2]:
detection_counts = pd.read_csv("data/most_detected_tags.csv")
dwell_loc = pd.read_csv("data/loc_code_detection_patterns.csv")
dwell_antenna = pd.read_csv("data/antenna_detection_patterns.csv")
location = pd.read_csv("data/location_lat_long.csv")

In [3]:
detection_counts

tag_id   count in_cleaning_table
0     989.001042  273253                no
1     989.001039  111089                no
2     989.001042   89789                no
3     989.001040   82246                no
4     989.001039   78683                no
...          ...     ...               ...
6781  989.001045       1                no
6782  989.001041       1                no
6783  989.001042       1                no
6784  989.001041       1                no
6785  989.001045       1                no

[6786 rows x 3 columns]

In [4]:
dwell_loc

tag_id loc_code  start_date    end_date  \
0      989.001007       7A  2020-09-26  2020-09-26   
1      989.001033       21  2020-09-27  2020-09-27   
2      989.001007       22  2020-10-03  2020-10-03   
3      989.001007      232  2020-10-23  2020-10-23   
4      989.001007      231  2020-10-23  2020-10-23   
...           ...      ...         ...         ...   
37242  989.002028      202  2025-01-31  2025-01-31   
37243  989.002028      201  2025-01-31  2025-01-31   
37244  989.001034      401  2025-02-01  2025-02-01   
37245  989.001034      402  2025-02-05  2025-02-05   
37246  989.002028      702  2025-02-06  2025-02-06   

                       start_time                    end_time  \
0      2020-09-26 10:36:39.030000  2020-09-26 10:38:44.320000   
1      2020-09-27 07:04:30.360000  2020-09-27 07:04:30.360000   
2      2020-10-03 04:12:49.280000  2020-10-03 04:12:49.280000   
3      2020-10-23 19:55:21.760000  2020-10-23 19:59:11.080000   
4      2020-10-23 19:59:21.450000  2020-10-23 20:05:58.390000   
...                           ...                         ...   
37242  2025-01-31 23:04:42.340000  2025-01-31 23:04:42.340000   
37243  2025-01-31 23:08:05.630000  2025-01-31 23:08:05.630000   
37244  2025-02-01 01:12:03.700000  2025-02-01 01:12:03.700000   
37245  2025-02-05 01:07:23.460000  2025-02-05 01:07:23.460000   
37246  2025-02-06 00:19:28.940000  2025-02-06 02:39:46.360000   

                   dwell_time  number_of_detections  
0      0 days 00:02:05.290000                     3  
1             0 days 00:00:00                     1  
2             0 days 00:00:00                     1  
3      0 days 00:03:49.320000                     2  
4      0 days 00:06:36.940000                     2  
...                       ...                   ...  
37242         0 days 00:00:00                     1  
37243         0 days 00:00:00                     1  
37244         0 days 00:00:00                     1  
37245         0 days 00:00:00                     1  
37246  0 days 02:20:17.420000                     3  

[37247 rows x 8 columns]

# Summaries

## For an individual tag

In [5]:
tag = 989.001006676104

print("Tag ID is:", tag)
print("Total number of detections:", detection_counts[detection_counts["tag_id"] == tag]["count"].iloc[0])
print("Found in cleaning table? ", detection_counts[detection_counts["tag_id"] == tag]["in_cleaning_table"].iloc[0])
print("First detected:", min(dwell_loc[dwell_loc.tag_id == tag]["start_date"]))
print("Last detected:", max(dwell_loc[dwell_loc.tag_id == tag]["end_date"]))

print("Details of each detection:")
tag_dwell = dwell_loc[dwell_loc.tag_id == tag]
tag_dwell

Tag ID is: 989.001006676104
Total number of detections: 78120
Found in cleaning table?  no
First detected: 2022-05-08
Last detected: 2023-05-16
Details of each detection:


tag_id loc_code  start_date    end_date                  start_time  \
1933  989.001007      402  2022-05-08  2022-06-28  2022-05-08 03:48:48.770000   
2593  989.001007      401  2022-06-28  2022-06-28  2022-06-28 18:40:58.040000   
2597  989.001007      402  2022-06-28  2022-06-28  2022-06-28 21:14:57.270000   
2599  989.001007      401  2022-06-29  2022-06-29  2022-06-29 00:13:00.200000   
2600  989.001007      402  2022-06-29  2022-06-29  2022-06-29 05:30:39.580000   
...          ...      ...         ...         ...                         ...   
7775  989.001007      402  2023-05-14  2023-05-15  2023-05-14 21:31:49.480000   
7827  989.001007      401  2023-05-15  2023-05-15  2023-05-15 20:34:21.440000   
7902  989.001007      402  2023-05-16  2023-05-16  2023-05-16 04:39:44.090000   
7903  989.001007      401  2023-05-16  2023-05-16  2023-05-16 04:44:27.310000   
7906  989.001007      402  2023-05-16  2023-05-16  2023-05-16 05:02:57.270000   

                        end_time               dwell_time  \
1933  2022-06-28 18:29:55.220000  51 days 14:41:06.450000   
2593  2022-06-28 19:06:18.400000   0 days 00:25:20.360000   
2597  2022-06-28 21:14:57.270000          0 days 00:00:00   
2599  2022-06-29 05:26:20.220000   0 days 05:13:20.020000   
2600  2022-06-29 10:43:41.870000   0 days 05:13:02.290000   
...                          ...                      ...   
7775  2023-05-15 20:28:24.390000   0 days 22:56:34.910000   
7827  2023-05-15 21:36:19.510000   0 days 01:01:58.070000   
7902  2023-05-16 04:39:44.090000          0 days 00:00:00   
7903  2023-05-16 04:44:27.310000          0 days 00:00:00   
7906  2023-05-16 05:18:59.830000   0 days 00:16:02.560000   

      number_of_detections  
1933                    14  
2593                     2  
2597                     1  
2599                     2  
2600                    10  
...                    ...  
7775                     2  
7827                     2  
7902                     1  
7903                     1  
7906                     2  

[343 rows x 8 columns]

# What does number of detections mean? What counts as one detection?

SELECT tag_id, datetime, detection_date, detection_time, loc_code <br>
FROM detections <br>
WHERE  tag_id = '989.001006676104' and loc_code = '402' <br>
ORDER BY datetime <br>

If the detections are close to reach other, one detection is every 5 minutes. Sometimes, we don't see them for a bit and then they are detected on the same antenna a few hours later.